libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -828,6 +828,45 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 				StorageByteArrayElement(m_context).storeValue(*type, _functionCall.location(), true);
 			break;
 		}
+		case FunctionType::Kind::ByteArrayPop:
+		case FunctionType::Kind::ArrayPop:
+		{
+			_functionCall.expression().accept(*this);
+			solAssert(function.parameterTypes().size() == 0, "");
+			// solAssert(!!function.parameterTypes()[0], "");
+			//TypePointer paramType = function.parameterTypes()[0];
+			shared_ptr<ArrayType> arrayType = make_shared<ArrayType>(DataLocation::Storage);
```

Comment:
This type is only correct in case of `ByteArrayPop`. The type of the array is a little hard to retrieve, though. You need to walk your way up the AST:

`dynamic_cast<MemberAccess const&>(_functionCall.expression()).expression().type()`

This can also fail in cases like: `var x = a.pop; x();` - but I think we should just disallow such constructs. There has been a PR recently that disallows `var x = block.blockhash; x();` in the same way I think.

Reply:
Makes sense. I was worried I'd need the type but wasn't sure how to get it. I'll see what I can do there